src/armory/Armory.js

```diff
@@ -1,108 +1,286 @@
-			scannerTwoOptions: [new OptionClass('','')],
-			scannerThreeOptions: [new OptionClass('','')],
-			jammerOptions: [new OptionClass('','')],
-			treadsOptions: [new OptionClass('','')],
-			singleUseItemsOne: [new OptionClass('','')],
-			singleUseItemsTwo: [new OptionClass('','')],
-			singleUseItemsThree: [new OptionClass('','')],
+		
+		this.state = {
+			selectedTank: null,
+			allTanks: [],
+			Inventory: [],
+			tankOptions: [],
+			chassis: [],
+			weapons: [],
+			scanners: [],
+			scannerAddons: [],
+			jammers: [],
+			treads: [],
+			items: [],
 		}
-		this.getFavoriteTank();
+
+		// Functions to get all user tanks and user inventory.
+		this.getTanks();
+		this.getInventory();
+	}
+
+	// Gets all user tanks and sets them to the state.
+	getTanks(): void {
+		const responsePromise = getAllUsersTanks();
+		
+		responsePromise.then (
+			response => response.json().then(data => {
+				if (response.status !== 200) {
+					console.log(response.status);
+					console.log(data.msg);
+					console.log(data);
+				}
+				else {
+					let allTanks: Array<Tank> = [];
+					for(const tank of data) {
+						allTanks.push(getTank(tank));
+					}
+					this.setState({allTanks: data});
+					// Set the first tank found as the selectedTank.
+					this.setState({selectedTank: getTank(data[0])});
+				}
+			})
+		)
+	}
+
+	// Gets all user inventory.
+	getInventory(): void {
+		const responsePromise = getUser();
+
+		responsePromise.then (
+			response => response.json().then(data => {
+				if(response.status !== 200) {
+					console.log(response.status);
+					console.log(data.msg);
+					console.log(data);
+				}
+				else {
+					this.setState({
+						inventory: data.inventory.tankComponents,
+						chassis: getOptionsOfType(data.inventory.tankComponents, 'chassis'),
+						weapons: getOptionsOfType(data.inventory.tankComponents, 'weapon'),
+						scanners: getOptionsOfType(data.inventory.tankComponents, 'scanner'),
+						scannerAddons: getOptionsOfType(data.inventory.tankComponents, 'scannerAddon'),
+						jammers: getOptionsOfType(data.inventory.tankComponents, 'jammer'),
+						treads: getOptionsOfType(data.inventory.tankComponents, 'treads'),
+						items: getOptionsOfType(data.inventory.tankComponents, 'item'),
+					});
+				}
+			})
+		)
+	}
+
+	// Converts camel case to title case.
+	toTitleCase(str: string): string {
+		// Check for undefined strings.
+		if(!str) {
+			return str;
+		}
+		
+		str = str.replace( /([A-Z])/g, " $1");
+		return str.charAt(0).toUpperCase() + str.slice(1);
+	}
+
+	// Find the tank via its name and set it to the selectedTank and in a Cookie for Casus.
+	changeSelectedTank(newTankName: string): void {
+		const newTank: Tank = getTank(this.state.allTanks.find(tank => tank.tankName === newTankName));
+		this.setState({ selectedTank: newTank});
+		const cookies = new Cookies();
+		cookies.set('tank', newTank);
+	}
+
+	// ALL UPDATES NEED API CALL TO UPDATE INVENTORY.
```

Comment:
Is this a work-in-progress?

Otherwise it doesn't look like it is doing much atm